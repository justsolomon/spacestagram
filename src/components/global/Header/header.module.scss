@import "styles/variables";

.header {
  margin: 1rem 7rem;

  &__title {
    font-size: $font-size--xxl;
    font-weight: $font-weight--bold;
    margin-bottom: 0.3rem;
  }

  &__attribution {
    color: $secondary-color;
  }

  @media #{$media-desktop-sm} {
    margin: 1rem 4rem;
  }

  @media #{$media-tablet} {
    margin: 1rem 6.5rem;
  }

  @media #{$media-mobile} {
    margin: 1rem 4rem;

    &__attribution {
      font-size: $font-size--sm;
    }
  }

  @media #{$media-mobile-sm} {
    margin: 1rem 1.5rem;
  }
}
