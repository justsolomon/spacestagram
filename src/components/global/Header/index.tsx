import styles from "./header.module.scss";

const Header = () => {
  return (
    <header className={styles["header"]}>
      <h1 className={styles["header__title"]}>Spacestagram</h1>
      <p className={styles["header__attribution"]}>
        Brought to you by NASA's Mars Rover Photos API
      </p>
    </header>
  );
};

export default Header;
