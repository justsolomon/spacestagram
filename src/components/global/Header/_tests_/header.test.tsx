import { render } from "@testing-library/react";
import Header from "..";

describe("<Header />", () => {
  it("should render the header content correctly", () => {
    const { asFragment } = render(<Header />);

    expect(asFragment).toMatchSnapshot();
  });
});
