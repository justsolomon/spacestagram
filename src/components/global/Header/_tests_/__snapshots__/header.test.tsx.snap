// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the header content correctly 1`] = `[Function]`;
